### PR TITLE
[ksm-core] Improve Job completion metrics

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Write here in detail how you have tested your changes
and instructions on how this should be tested in QA.

Describe or link instructions to set up environment 
for testing, if the process requires more than just
running the agent on one of the supported platforms.

### Checklist

- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

File: pkg/collector/corechecks/cluster/ksm/kubernetes_state_transformers.go [hunk 1]
```diff
@@ -298,11 +298,21 @@ func cronJobLastScheduleTransformer(s aggregator.Sender, name string, metric ksm
 
 // jobCompleteTransformer sends a service check based on kube_job_complete
 func jobCompleteTransformer(s aggregator.Sender, name string, metric ksmstore.DDMetric, hostname string, tags []string) {
+	for i, tag := range tags {
+		if tag == "condition:true" {
+			jobMetric(s, metric, ksmMetricPrefix+"job.completion.succeeded", hostname, append(tags[:i], tags[i+1:]...))
```
Comment (davidor) [inline pkg/collector/corechecks/cluster/ksm/kubernetes_state_transformers.go:303]: Should there be a `break` here?

Comment (davidor) [reply pkg/collector/corechecks/cluster/ksm/kubernetes_state_transformers.go:303]: Also, should we update the description of the function to reflect the new reporting?

Comment (L3n41c) [reply pkg/collector/corechecks/cluster/ksm/kubernetes_state_transformers.go:303]: Definitively yes.
I’ve added the `break` and updated the function comment.

**Comment by davidor** [comment]
Looks good to me, but even though it's a small change, I'd add a test to make sure we don't break this.

**Review by clamoriniere — APPROVED** [review]

**Comment by L3n41c** [comment]
Tests added!

**Pull request merged**